lib/build.ts

```diff
@@ -328,32 +333,19 @@ async function checkAllowDockerPlatformHandling(
 		return true;
 	} else if (imagesWithPlatformSupport.length > 0) {
 		// Mixed images
-		console.log(
-			'Warning: Dockerfile contains a mixes images that require a platform selection with ones that do not support it.\n' +
-				'The following images do not support platform selection:\n-' +
-				imagesWithoutPlatformSupport.join('\n-') +
-				'\n\n' +
-				'The following images require platform selection:\n-' +
-				imagesWithPlatformSupport.join('\n-') +
-				'\n\n' +
-				'To avoid a possible build error, the CLI has disabled platform selection. ' +
-				'As a result, the architecture of the machine where the Docker Engine ' +
-				'is running will be used to select the platform when pulling upstream images ' +
-				'which may result in "exec format error" at runtime. To avoid this warning, ' +
-				'update or replace the images that do not support platform selection.',
-		);
-	} else {
-		// Only images without platform support
-		console.log(
-			'Warning: Dockerfile contains a images do not support platform selection.\n-' +
-				imagesWithoutPlatformSupport.join('\n-') +
-				'\n\n' +
-				'To avoid a possible build error, the CLI has disabled platform selection. ' +
-				'As a result, the architecture of the machine where the Docker Engine ' +
-				'is running will be used to select the platform when pulling upstream images ' +
-				'which may result in "exec format error" at runtime. To avoid this warning, ' +
-				'update or replace the images that do not support platform selection.',
-		);
+		warn(`\
+Service '${task.serviceName}':
+  Multi-stage Dockerfile found with a mix of base images that require
+  CPU architecture selection and base images that do not support it.
+  The following base images do not support CPU architecture selection:
+  - ${imagesWithoutPlatformSupport.join('\n  - ')}
+  The following base images require CPU architecture selection:
+  - ${imagesWithPlatformSupport.join('\n  - ')}
+  As a result, the CPU architecture of the machine where the Docker Engine
+  is running will be used to select base images that require architecture
+  selection. This may result in incorrect architecture selection and "exec
+  format error" at runtime. To avoid this warning, update or replace base
+  images that do not support architecture selection.`);
```

Comment:
Here's how this ends up looking with the CLI -- the lines prefixed with `[Warn]`:
```
$ ./bin/balena build t -A armv7hf -d raspberrypi3 --nocache --pull --debug
...
[Info]    Building for armv7hf/raspberrypi3
[Info]    Docker Desktop detected (daemon architecture: "x86_64")
[Info]      Docker itself will determine and enable architecture emulation if required,
[Info]      without balena-cli intervention and regardless of the --emulated option.
[Debug]   Found build tasks:
[Debug]       main: build [.]
[Debug]   Resolving services with [raspberrypi3|armv7hf]
[Debug]   Found project types:
[Debug]       main: Standard Dockerfile
[Debug]   Prepared tasks; building...
[Warn]    Service 'main':
[Warn]      Multi-stage Dockerfile found with a mix of base images that require
[Warn]      CPU architecture selection and base images that do not support it.
[Warn]      The following base images do not support CPU architecture selection:
[Warn]      - balenalib/raspberrypi3-node:14-buster-run
[Warn]      The following base images require CPU architecture selection:
[Warn]      - debian:10
[Warn]      As a result, the CPU architecture of the machine where the Docker Engine
[Warn]      is running will be used to select base images that require architecture
[Warn]      selection. This may result in incorrect architecture selection and "exec
[Warn]      format error" at runtime. To avoid this warning, update or replace base
[Warn]      images that do not support architecture selection.
[Build]   main Step 1/5 : FROM balenalib/raspberrypi3-node:14-buster-run as base
[Build]   main  ---> e074e38a89ba
```